src/analysis/function_parameters.rs

```diff
@@ -262,6 +261,25 @@ pub fn analyze(
             ind_rust = None;
         }
 
+        let mut trans_nullable = false;
+        let type_ = env.type_(par.typ);
+        let to_glib_extra = if par.instance_parameter || !*nullable ||
+                               (!type_.is_interface() && !type_.is_class()) {
+            String::new()
+        } else {
+            trans_nullable = *nullable;
+            match type_.maybe_ref() {
+                Some(&library::Class { final_type, ..}) => {
+                    if final_type {
```

Comment:
IMHO final_type better moved to library and return false for all other types,
bad that this don't help optimize this condition as we need set `trans_nullable` too

Reply:
Fine by me!